### ISSUE TITLE
NewClient() seems to be returning nil without throwing an error

### ISSUE THREAD

**Issue by Hoid** [andygrunwald/go-jira#106]
Hi all,

I have a problem where attempting to get an issue's info throws a nil pointer exception. The error is the following: 
`runtime error: invalid memory address or nil pointer dereference: errorString`

Here's where I'm accessing the jiraClient, inside the OnIntent method:
```
log.Println("OpenTicketIntent triggered")

issue, issueResponse, err := jiraClient.Issue.Get("DUN-455", nil)
fmt.Println(issue)
fmt.Println(issueResponse)
if err != nil {
	log.Println(err.Error())
}
i := fmt.Sprintf("%s: %+v\n", issue.Key, issue.Fields.Summary)

speechText := fmt.Sprintf("i = %s", i)
```

It is on this line that I'm getting the error:
`issue, issueResponse, err := jiraClient.Issue.Get("DUN-455", nil)`

Here is where I'm initializing jiraClient, in main:
```
jiraClient, err := jira.NewClient(nil, "https://xxxxxx.atlassian.net/")
if err != nil {
	fmt.Println(err.Error())
	panic(err)
}
fmt.Print("jiraClient: ")
fmt.Println(jiraClient)
jiraClient.Authentication.SetBasicAuth("xxxxx", "xxxxx")
lambda.Start(Handle)
```

I'm sure that the line `log.Println("OpenTicketIntent triggered")` is getting triggered because I see that in the console output. What I _don't_ see in the console is where I'm printing the jiraClient with a prefix string:
```
fmt.Print("jiraClient: ")
fmt.Println(jiraClient)
```
But in addition, I'm not getting an error thrown in main! My program goes all the way into OnIntent and into OpenTicketIntent before trying to access the jiraClient, which was initialized in main. 

This is making me think that NewClient() is returning a nil response for the client without throwing an error. Am I missing something here?

**Comment by rbriski** [comment]
I tried to reproduce this with a working JIRA site but couldn't.  Also, I don't understand where `OnIntent` and `OpenTicketIntent` are coming from.  Are these functions in your own code?  

Are you sure the configuration for the client is correct?  Are you able to provide any more information or create a failing test?

**Comment by Hoid** [comment]
`OnIntent` comes from [here, a go port of the alexa skills kit](https://github.com/ericdaugherty/alexa-skills-kit-golang) and `OpenTicketIntent` comes from my Interaction Model in the Amazon Alexa Skills Kit configuration. It's an intent that I've configured and it's just a case in a switch statement in my code
that gets triggered.

I'm sure that I've tried all of the different configurations I can think of, including the ones in your examples. I'm not sure what more information I could provide. Is there something you are looking for specifically?

**Comment by rbriski** [comment]
@Hoid , you're describing an error for a variable that I can't see in either this repo, `go-jira`, or the alexa skills kit that you reference.  The error you mention is:
`runtime error: invalid memory address or nil pointer dereference: errorString`

Where is `errorString`?  Why isn't it being set?  Where is it supposed to be set?  You're referencing code that no one else has access to.  

If you want to post a gist or include a link to the repo you're talking about, someone might be able to help.

**Comment by Hoid** [comment]
[main.zip](https://github.com/andygrunwald/go-jira/files/1734164/main.zip)
Here's the code I'm working with

**Comment by rbriski** [comment]
You just zipped up a binary file named `main`.  Please create a repository and check it in on Github so I can look at it.

**Comment by Hoid** [comment]
Oops sorry. I've included the actual .go file now. I don't want to put this code on github, though, because I don't want it to be public for long and it's easier to delete a comment than a repo.
[main.zip](https://github.com/andygrunwald/go-jira/files/1734257/main.zip)

**Comment by rbriski** [comment]
I extracted the code that actually uses `go-jira` here: https://gist.github.com/rbriski/6637af24c0c5447410175a7579618f75

If you fill in your credentials and JIRA site, it works fine.  Any issue you're running into is with your code or some other library.

I tried to run the whole thing but the Alexa library your using has made some backward incompatible changes since you last pulled your dependencies.  I couldn't even get `go get` to work.
